packages/library-utils/src/library_account_type.rs

```diff
@@ -0,0 +1,273 @@
+        match self {
+            LibraryAccountType::Addr(addr) => Ok(addr.to_string()),
+            LibraryAccountType::AccountId(_) | LibraryAccountType::LibraryId(_) => Err(
+                StdError::generic_err("LibraryAccountType must be an address"),
+            ),
+        }
+    }
+
+    /// Returns the address as Addr if it is an address, otherwise returns an error
+    pub fn to_addr(&self, api: &dyn cosmwasm_std::Api) -> StdResult<Addr> {
+        match self {
+            LibraryAccountType::Addr(addr) => api.addr_validate(addr),
+            LibraryAccountType::AccountId(_) | LibraryAccountType::LibraryId(_) => Err(
+                StdError::generic_err("LibraryAccountType must be an address"),
+            ),
+        }
+    }
+
+    /// There are cases where a library config expects a string, but we still want to use the
+    /// id replacement functionality of the manager.
+    /// Using this function will use a placeholder that can be replaced by the manager
+    /// to the instantiated address
+    pub fn to_raw_placeholder(&self) -> String {
+        let value = match self {
+            // If its an address, we can use it directly
+            LibraryAccountType::Addr(addr) => return addr.to_string(),
+            LibraryAccountType::AccountId(id) => id.to_string(),
+            LibraryAccountType::LibraryId(_) => {
+                panic!("Only accounts can use raw_placeholder functionality")
+            }
+        };
+
+        format!("{}:{}", LIBRARY_ACCOUNT_RAW_PLACEHOLDER, value)
+    }
+}
+
+impl From<&Addr> for LibraryAccountType {
+    fn from(addr: &Addr) -> Self {
+        LibraryAccountType::Addr(addr.to_string())
+    }
+}
+
+impl From<&str> for LibraryAccountType {
+    fn from(input: &str) -> Self {
+        if input.starts_with("{\"|account_id|\":") {
+            LibraryAccountType::AccountId(
+                input
+                    .trim_start_matches("{\"|account_id|\":")
+                    .trim_end_matches("}")
+                    .parse()
+                    .expect("Failed parsing account_id into LibraryAccountType"),
+            )
+        } else if input.starts_with("{\"|library_id|\":") {
+            LibraryAccountType::LibraryId(
+                input
+                    .trim_start_matches("{\"|library_id|\":")
+                    .trim_end_matches("}")
+                    .parse()
+                    .expect("Failed parsing library_id into LibraryAccountType"),
+            )
+        } else if input.starts_with("{\"|library_account_addr|\":\"") {
+            LibraryAccountType::Addr(
+                input
+                    .trim_start_matches("{\"|library_account_addr|\":\"")
+                    .trim_end_matches("\"}")
+                    .parse()
+                    .expect("Failed parsing addr into LibraryAccountType"),
+            )
+        } else {
+            LibraryAccountType::Addr(input.to_string())
+        }
```

Comment:
some comment here would be helpful about why in case the `if input.starts_with("{\"|library_account_addr|\":\"")` check fails we still default to `LibraryAccountType::Addr`?